pstoreds/addr_book.go

```diff
@@ -402,14 +401,12 @@ func (ab *dsAddrBook) addrs(p peer.ID, includeUncertified bool) []ma.Multiaddr {
 	pr.RLock()
 	defer pr.RUnlock()
 
-	addrs := make([]ma.Multiaddr, 0, len(pr.Addrs))
+	var addrs []ma.Multiaddr
```

Comment:
why not presize to the correct size? We call this quite often, so this could make a difference.

Reply:
Yup, agree. I'm merging into the base branch (which is another feature branch), but we can definitely presize this.

Reply:
```suggestion
    // TODO: presize this slice.
	var addrs []ma.Multiaddr
```